src/reaction.jl

```diff
@@ -134,19 +138,19 @@ Notes:
 - The three-argument form assumes all reactant and product stoichiometric coefficients
   are one.
 """
-struct Reaction{S, T}
+struct Reaction{T}
     """The rate function (excluding mass action terms)."""
     rate::Any
     """Reaction substrates."""
-    substrates::Vector
+    substrates::Vector{Any}
     """Reaction products."""
-    products::Vector
+    products::Vector{Any}
     """The stoichiometric coefficients of the reactants."""
-    substoich::Vector{T}
+    substoich::Vector
     """The stoichiometric coefficients of the products."""
-    prodstoich::Vector{T}
+    prodstoich::Vector
     """The net stoichiometric coefficients of all species changed by the reaction."""
-    netstoich::Vector{Pair{S, T}}
+    netstoich::Vector{Pair{Any, T}}
```

Comment:
When I first started looking at this I realised that we didn't typed everything as stringently as we could (i.e. first I figured `substrates::Vector{R}` would be better than just `substrates::Vector`, and then I realised we actually knew the type so could insert it. I think I was still in the old mode when we just narrowed down things as much as possible. I agree that if there is any doubt, there is no need. I really am not familiar with the impact of typing, I pretty much started doing it because you where doing it, but not based on any actual insight. If you have any doubts about any typing I 100% leave it to you to make that decision.

I also properly messed-up when I tried to revert to use the old typing system. That was not my intention but a really bad mistake, I am sorry about that.